cardano-api/src/Cardano/Api/TxBody.hs

```diff
@@ -1470,6 +1470,17 @@ deserialiseShelleyBasedTxBody era bs =
 
       case len of
         -- Backwards compat for pre-Alonzo era tx body files
+        2 -> do
+          txbody     <- fromCBOR
+          txmetadata <- CBOR.decodeNullMaybe fromCBOR
+          pure $ CBOR.Annotator $ \fbs ->
+            ShelleyTxBody
+              era
+              (CBOR.runAnnotator txbody fbs)
+              [] -- scripts
+              (CBOR.runAnnotator (pure TxBodyNoScriptData) fbs)
+              (fmap (`CBOR.runAnnotator` fbs) txmetadata)
+              (CBOR.runAnnotator (pure TxScriptValidityNone) fbs)
```

Comment:
If you're copying and pasting anyway, don't add extra unnecessary changes too. It makes it harder to compare to see that they're the same.
```suggestion
              (flip CBOR.runAnnotator fbs (return TxBodyNoScriptData))
              (fmap (flip CBOR.runAnnotator fbs) txmetadata)
              (flip CBOR.runAnnotator fbs (return TxScriptValidityNone))
```
If you think the existing code is poor style then change that too. Or share the code so there's only a single copy and make that the nice style. But don't copy with minor unnecessary changes, that way lies madness.

Reply:
Sorry, I had to simplify this to get into it.

Reply:
done